File: src/main/resources/static/entity/support.js
export class Support{
    id;
    status;
    subject;
    content;
    response;
    date;
    //list
    restaurant;

    constructor(id, status, subject, content, response, date, restaurant) {
        this.id = id;
        this.status = status;
        this.subject = subject;
        this.content = content;
        this.response = response;
        this.date = date;
        this.restaurant = restaurant;
    }
}

File: src/main/java/com/restaurant/restaurantordersystem/entity/Support.java
package com.restaurant.restaurantordersystem.entity;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Data
@Entity
@Table
//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class , property = "id")
@JsonIdentityReference
public class Support {
    /*have 2 different statuses:
        0 : not responded yet.
        1: responded.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer status;
    private String subject;
    private String content;
    private String response;
    private Date date;

    @ManyToOne
    private Restaurant restaurant;

    transient private Auth admin;

    public void seen(){
        setStatus(1);
    }

}
